doc/how_to/custom_components/esm/custom_layout.md

```diff
@@ -157,71 +171,52 @@ Lets verify the layout will automatically update when the `object` is changed.
 
 ::::{tab-set}
 
-:::{tab-item} `Viewer`
+:::{tab-item} `JSComponent`
 
 ```{pyodide}
-html = pn.pane.Markdown("A **markdown** pane!", name="Markdown")
-radio_button_group = pn.widgets.RadioButtonGroup(
-    options=["Dial", "Markdown"],
-    value="Dial",
-    name="Select the object to display",
-    button_type="success", button_style="outline"
-)
+split_js.right=pn.pane.Markdown("Hi. I'm a `Markdown` pane replacing the `CodeEditor` widget!", sizing_mode="stretch_both")
```

Comment:
When updating one child it seems like the other child is also updating. That won't look nice and cannot be efficient.

https://github.com/user-attachments/assets/48037e2d-2b5e-4085-9ba4-aecac33d30a6

```python
import panel as pn

from panel.custom import Child, JSComponent

CSS = """
.split {
    display: flex;
    flex-direction: row;
    height: 100%;
    width: 100%;
}

.gutter {
    background-color: #eee;
    background-repeat: no-repeat;
    background-position: 50%;
}

.gutter.gutter-horizontal {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAeCAYAAADkftS9AAAAIklEQVQoU2M4c+bMfxAGAgYYmwGrIIiDjrELjpo5aiZeMwF+yNnOs5KSvgAAAABJRU5ErkJggg==');
    cursor: col-resize;
}
"""


class SplitJS(JSComponent):

    left = Child()
    right = Child()

    _esm = """
    import Split from 'https://esm.sh/split.js@1.6.5'

    const splitDiv = document.createElement('div');
    splitDiv.className = 'split';

    const split0 = document.createElement('div');
    splitDiv.appendChild(split0);

    let split1 = document.createElement('div');
    splitDiv.appendChild(split1);

    Split([split0, split1])

    export function render({ model }) {
        split0.append(model.get_child("left"))
        split1.append(model.get_child("right"))
      return splitDiv
    }"""

    _stylesheets = [CSS]


pn.extension("codeeditor")

split_js = SplitJS(
    left=pn.widgets.CodeEditor(
        value="Left!",
        sizing_mode="stretch_both",
        margin=0,
        theme="monokai",
        language="python",
    ),
    right=pn.widgets.CodeEditor(
        value="right",
        sizing_mode="stretch_both",
        margin=0,
        theme="monokai",
        language="python",
    ),
    height=500,
    sizing_mode="stretch_width",
)
split_js.servable()

def update():
    print("update")
    split_js.right=pn.pane.Markdown("Hi. I'm a `Markdown` pane replacing the `CodeEditor` widget!", height=500, sizing_mode="stretch_width")

pn.state.add_periodic_callback(update, period=1000, count=10)
```

Reply:
Fixed in https://github.com/holoviz/panel/pull/7163

Note though that initializing DOM nodes in the module (rather than the render function) is something I'd consider an absolute no-no:

```python
    const splitDiv = document.createElement('div');
    splitDiv.className = 'split';

    const split0 = document.createElement('div');
    splitDiv.appendChild(split0);

    let split1 = document.createElement('div');
    splitDiv.appendChild(split1);

    Split([split0, split1])
```